server/models/productModel.js

```diff
@@ -37,4 +38,12 @@ const ProductModel = db.define("Product", {
   },
 });
 
+NotificationModel.belongsTo(ProductModel, {
+  as: "Product",
+  foreignKey: {
+    name: 'productId',
+    allowNull: false
+  }
+});
+
 module.exports = ProductModel;
```

Comment:
so a notification isn't actually tied to a product. 
The notification should have an id, title, image, description, and whether it is read. This way we can use it for many different things.

Reply:
A notification has a product, so a Notification Model would be "id" which is the id of the product and the "productId" which would reference the current information of the "productId". Now that I think about it, it seems like there would be a bug wherein they would all be referencing the same versions of the product.

Reply:
So now that I think about it, the reason the notification shouldn't link to a product id, is because say we have this situation:
1. We add `Laptop` to our list. Original name is `"Asus X"` and original price is `$20`
2. Price of `Laptop` drops to `$10`
3. We generate a notification telling user: `Laptop named Asus X dropped from $20 to $10`
4. 2 days later Price of `Laptop` drops again to `$5` and name is changed to `"Asus Y"`
5. We generate a notification that tells the user: `Laptop named Asus Y dropped from $10 to $5`

Now if we link the notification to a product id, and we are looking at our notification history, both of the notifications will say `Laptop named Asus Y dropped from $10 to $5`
So I think the notifications should have copies of this information. So they should just have `picture`, `type`, `message`
That way we can have different notifications like:
- "`Laptop named Asus Y dropped from $10 to $5`" with picture of the laptop and type: price change
- "User Emily has followed you" with profile picture of user and type: user update
- "You have received a message" with type: messages
etc.